torch/utils/_sympy/solve.py

```diff
@@ -0,0 +1,120 @@
+from typing import Dict, Optional, Tuple, Type
+
+import sympy
+
+from torch.utils._sympy.functions import FloorDiv
+
+
+# Tries to simplify 'expr', so as to leave only 'thing' in the left-hand side.
+# Returns a tuple of:
+#   1. The simplified expression
+#   2. The expression on the right-hand side
+# Returns 'None' if it can't reach a state where the only thing in the left
+# hand side is 'thing'.
+def try_solve(
```

Comment:
I'd suggest thinking if there's some good testing strategy for this function directly.

Reply:
I guess we could, at least, try to cover all execution paths. Do you think that would be enough?
Maybe we could also take a few examples that appear when running some models. What do you think?

Reply:
Could we have a list of inputs that exercises one and only one of the paths and then prove with Z3 that the result is equivalent?